src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -288,6 +292,14 @@ internal static bool DoSslHandshake(SafeSslHandle context, byte[] recvBuf, int r
                 {
                     sendCount = BioRead(context.OutputBio, sendBuf, sendCount);
                 }
+                catch (Exception e)
+                {
+                    // If we already have handshake exception, ignore any exception from BioRead().
+                    if (handshakeException == null)
+                    {
+                        handshakeException = e;
+                    }
```

Comment:
Does any ERR_clear_error() or any other "don't leave the thread with a pending error" cleanup need to happen, or did that already happen in BioRead?

Reply:
I'm not sure if it was needed but I updated code to do it on failure before BioRead().